frappe/core/report/database_storage_usage_by_tables/database_storage_usage_by_tables.js

```diff
@@ -3,4 +3,49 @@
 
 frappe.query_reports["Database Storage Usage By Tables"] = {
 	filters: [],
+	onload: function (report) {
+		report.page.add_inner_button(
+			__("Optimize"),
+			function () {
+				let d = new frappe.ui.Dialog({
+					title: "Optimize Doctype",
+					fields: [
+						{
+							label: "Select a DocType",
+							fieldname: "doctype_name",
+							fieldtype: "Link",
+							options: "DocType",
+							get_query: function () {
+								return {
+									filters: { issingle: ["=", false], is_virtual: ["=", false] },
+								};
+							},
+						},
+					],
+					size: "small",
+					primary_action_label: "Optimize",
+					primary_action(values) {
+						frappe.call({
```

Comment:
Should it be run in the background? The request could timeout for larger tables otherwise